File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(),people.end());
        int boats=0, start=0, end=people.size()-1 ;
        while(start <= end){
            if(people[start] + people[end] <= limit){  //Both can come in a single boat
                start++ ;
                end-- ;
            }
            else end-- ;    // Ommiting last one (being greedy)
            boats++;
        }
        return boats;        
    }
};

File: 0881-boats-to-save-people/0881-boats-to-save-people.ts
function numRescueBoats(people: number[], limit: number): number {
    people.sort((a,b)=>a-b);
    let boats=0;
    let start=0;
    let end = people.length - 1;
    while(start<=end){
        if(people[start] + people[end] <= limit){
            boats++;
            start++;
            end--;
        }
        else{
            end--;
            boats++;
        } 
    }
    return Number(boats);
};